server/api/socket/marker.py

```diff
@@ -0,0 +1,36 @@
+import auth
+from app import app, logger, sio, state
+from models import Marker
+from models.db import db
+
+
+@sio.on("Marker.New", namespace="/planarally")
+@auth.login_required(app, sio)
+async def new_marker(sid, data):
+    sid_data = state.sid_map[sid]
+    user = sid_data["user"]
+    location = sid_data["location"]
+
+    marker = Marker.get_or_none(shape=data)
+
+    if marker is not None:
+        return
+
+    Marker.create(
+        shape=data,
+        user=user,
+        location=location
+    )
+
+@sio.on("Marker.Remove", namespace="/planarally")
+@auth.login_required(app, sio)
+async def delete_marker(sid, uuid):
+    sid_data = state.sid_map[sid]
+    user = sid_data["user"]
+    location = sid_data["location"]
+
+    marker = Marker.get_or_none(uuid=uuid)
```

Comment:
```suggestion
    marker = Marker.get_or_none(shape=uuid, user=user)
```
the `uuid` field no longer exists and additionally you need to explicitly check the user, or you could accidentally remove someone else's marker of the same shape.

Reply:
Done